nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,739 @@
+        header_ornt = asstr(trk_reader.header[Field.VOXEL_ORDER])
+        affine_ornt = "".join(nib.orientations.aff2axcodes(trk_reader.header[Field.VOXEL_TO_RASMM]))
+        header_ornt = nib.orientations.axcodes2ornt(header_ornt)
+        affine_ornt = nib.orientations.axcodes2ornt(affine_ornt)
+        ornt = nib.orientations.ornt_transform(header_ornt, affine_ornt)
+        M = nib.orientations.inv_ornt_aff(ornt, trk_reader.header[Field.DIMENSIONS])
+        affine = np.dot(M, affine)
+
+        # Applied the affine found in the TRK header.
+        # voxel -> rasmm
+        affine = np.dot(trk_reader.header[Field.VOXEL_TO_RASMM], affine)
+
+        # Find scalars and properties name
+        data_per_point_slice = {}
+        if trk_reader.header[Field.NB_SCALARS_PER_POINT] > 0:
+            cpt = 0
+            for scalar_name in trk_reader.header['scalar_name']:
+                scalar_name = asstr(scalar_name)
+                if len(scalar_name) == 0:
+                    continue
+
+                # Check if we encoded the number of values we stocked for this scalar name.
+                nb_scalars = 1
+                if scalar_name[-2] == '\x00' and scalar_name[-1] != '\x00':
+                    nb_scalars = int(np.fromstring(scalar_name[-1], np.int8))
+
+                scalar_name = scalar_name.split('\x00')[0]
+                data_per_point_slice[scalar_name] = slice(cpt, cpt+nb_scalars)
+                cpt += nb_scalars
+
+            if cpt < trk_reader.header[Field.NB_SCALARS_PER_POINT]:
+                data_per_point_slice['scalars'] = slice(cpt, trk_reader.header[Field.NB_SCALARS_PER_POINT])
+
+        data_per_streamline_slice = {}
+        if trk_reader.header[Field.NB_PROPERTIES_PER_STREAMLINE] > 0:
+            cpt = 0
+            for property_name in trk_reader.header['property_name']:
+                property_name = asstr(property_name)
+                if len(property_name) == 0:
+                    continue
+
+                # Check if we encoded the number of values we stocked for this property name.
+                nb_properties = 1
+                if property_name[-2] == '\x00' and property_name[-1] != '\x00':
+                    nb_properties = int(np.fromstring(property_name[-1], np.int8))
+
+                property_name = property_name.split('\x00')[0]
+                data_per_streamline_slice[property_name] = slice(cpt, cpt+nb_properties)
+                cpt += nb_properties
+
+            if cpt < trk_reader.header[Field.NB_PROPERTIES_PER_STREAMLINE]:
+                data_per_streamline_slice['properties'] = slice(cpt, trk_reader.header[Field.NB_PROPERTIES_PER_STREAMLINE])
+
+
+        if lazy_load:
+            def _read():
+                for pts, scals, props in trk_reader:
+                    data_for_points = {k: scals[:, v] for k, v in data_per_point_slice.items()}
+                    data_for_streamline = {k: props[v] for k, v in data_per_streamline_slice.items()}
```

Comment:
For Python 2.6 compatibility:

    data_for_points = dict((k, scals[:, v]) for k, v in data_per_point_slice.items())
    data_for_streamline = dict((k, props[v]) for k, v in data_per_streamline_slice.items())